README.md

```diff
@@ -50,7 +50,20 @@ llm_chain:
             system_prompt: 'You are a helpful assistant that can answer questions.' # The default system prompt of the chain
             include_tools: true # Include tool definitions at the end of the system prompt 
             tools:
-                - 'PhpLlm\LlmChain\Chain\ToolBox\Tool\SimilaritySearch'
+                # Referencing a service with #[AsTool] attribute
+                - 'PhpLlm\LlmChain\Chain\Toolbox\Tool\SimilaritySearch'
+                
+                # Referencing a service without #[AsTool] attribute
+                - service: 'App\Chain\Tool\CompanyName'
+                  name: 'company_name'
+                  description: 'Provides the name of your company'
+                  method: 'foo' # Optional with default value '__invoke'
+                
+                # Referencing a chain => chain in chain 🤯
+                - service: 'llm_chain.chain.research'
+                  name: 'wikipedia_research'
+                  description: 'Can research on Wikipedia'
+                  is_chain: true
```

Comment:
While thinking about this, do we really need this? We know `service` is a `ChainInterface` 🤔

Reply:
another thing of writing could be:
```diff
- service: 'llm_chain.chain.research'
+ chain: 'llm_chain.chain.research'
name: 'wikipedia_research'
description: 'Can research on Wikipedia'
- is_chain: true
```

Reply:
but I think I would just remove the `is_chain`

Reply:
i tried to omit that thing, but failed :cry:

Reply:
but why, don't we know at that time, that the service is an instance of `ChainInterface`?

Reply:
wait a sec

Reply:
waiting.... ⏳

Reply:
![image](https://github.com/user-attachments/assets/e3613a11-fff6-4bf1-ab08-6caeb0c34164)

![image](https://github.com/user-attachments/assets/20070595-11ca-4604-90fb-9660eeda8b93)

Reply:
when i remove the `clock` stuff up there it works - at least for the chain. but that was the reason i avoided using `findDefinition`  - not sure why the `clock` is not found but later on can be resolved - i guess it's not available at this point

Reply:
but we would have the same issue, if we're referencing a chain that only will be registered below :man_shrugging:

Reply:
I mean, yes, it is no service available:
![CleanShot 2025-03-20 at 23 33 08@2x](https://github.com/user-attachments/assets/922326c2-d88c-415f-8a1d-007f7709feb7)

it is just a string, that said, what if you use hasDefinition or (getDefinition + try/catch) and IF it has a service, checking for the interface, like you did in the next line?

Reply:
that wouldn't solve the dependency of the order. with that `hasDefinition` check we would introduce the bug that a service that is a chain, but defined later in the configuration, would not be identified as chain.

Reply:
i'm with you on this, but i just don't know how :laughing:

Reply:
Alright then, but I can also set is_chain true and add anything else in service then, but we can communicate that with a clear error I guess

Reply:
wait a sec

Reply:
waiting again ... ⏳ 😄

Reply:
quite in your face, yes

![image](https://github.com/user-attachments/assets/b3f58673-a8e5-4715-80e1-e553c7b2ed2b)